### DISCUSSION TITLE
Can't Push Commit

### DISCUSSION THREAD

**Discussion by MoniquePipkin** [community/community#36487] [Other Feature Feedback, Questions, & Ideas] [answered]
I am currently using R 4.2.0 on the Mac OS Monteray. I recently cloned my repo to my local Rstudio and attempted to PUSH the committed changes to my GitHub but, I have received the following message.

>>> /usr/bin/git push origin HEAD:refs/heads/main
error: cannot run rpostback-askpass: No such file or directory
fatal: could not read Username for 'https://github.com': Device not configured

Any help appreciated!

**Comment by antonblr** [comment]
Hi @MoniquePipkin - please checkout this answer as it outlines the issue and possible solutions https://stackoverflow.com/a/40274775

**Comment by ossanna16** [comment]
Hey there @MoniquePipkin and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by MoniquePipkin** [comment, marked as answer]
Thanks @antonblr! I used your link as a jumping off point. I am now able to Push back to github. Here is a list of what I did. I am not sure which aspect or combonation of this allowed me to resolve the error, and so I have just included the entire process.

-I found [this page](https://stackoverflow.com/questions/22147574/github-fatal-could-not-read-username-for-https-github-com-no-such-file-o), which was helpful. From this link, I saw that one of the possible solutions from the above link said that this error occurred when cloning the repo from HTTPS and not SSH URL. I was pretty sure that I previously had a SSH key. However, there was a recent modification to GitHub earlier this year which may have reset or disconnected these things.

- First, I checked to see if I had an SSH key with the [ instructions on this page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys) 
- I didn't have any SSH keys that ended in ".pub" which is necessary for GITHUB
- Second, I generated a new SSH key [with the instructions on this page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent). I didn't add the SSH agent.
- Third, I added my SSH key to my GITHUB [following the instructions on this page.](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
- Forth, I tested by SSH key which said I was authenticated [following the instructions on this page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection)
- Fifth, Using I went onto my computer and added the associated .R, .csv files and README files to a backup folder.
- Sixth, I deleted the repo from my GITHUB and the associated PROJ file from my computer
- Seventh, I created a new repo on GITHUB.
- Eighth, I created a new project via Version Control and clone the repo to my computer using the SSH credentials instead of the HTTPS credentials
- Ninth, I copied over the necessary files: .R and .csv to the new project folder on my computer into the newly created project folder
- Tenth, I opened up the new project in R studio. I saved, Diff, Commit and PUSHed the changes into Github
- It all worked!!!
